### ISSUE TITLE
Move Documentation to Github Page

### ISSUE THREAD

**Issue by hayleycd** [google/osv-scanner#201]
Hello!

This issue is to discuss moving the documentation out of the readme and into a github page. 

There are a few issues I want to iron out before opening a PR. 

1). What branch do we want to use to build the documentation? We can use "last-stable" or create a doc specific branch. Ideally, this branch will include up to date information, but not any information on features that are not yet available. We can also just use main, but I'm trying to avoid problems like the one that cropped up in #196. I'm not sure what the easiest thing to do from a maintenance standpoint and would like your thoughts. 

2). I am assuming the github page should be https://google.github.io/osv-scanner please let me know if a different url is preferred. 

3). Upon merge, I will need someone with access to the project settings to set up the github page. 

4). Are there any other questions or concerns that we can work out? 

For reference, [this](https://github.com/hayleycd/osv-scanner/tree/doc-structure/docs) is the doc folder of the branch I am working on, and you can see how the docs currently render [here](https://hayleycd.github.io/osv-scanner/)

**Comment by another-rex** [comment]
> 1). What branch do we want to use to build the documentation? ...

I think instead of having a last-stable branch that is fixed to the release commit, we can update/cherry pick that branch with documentation updates as necessary until the next release. 

@oliverchang How is this done with oss-fuzz?

> 2). I am assuming the github page should be https://google.github.io/osv-scanner please let me know if a different url is preferred.

That url looks good to me.

**Comment by oliverchang** [comment]
> > 1). What branch do we want to use to build the documentation? ...
> 
> I think instead of having a last-stable branch that is fixed to the release commit, we can update/cherry pick that branch with documentation updates as necessary until the next release.
> 
> @oliverchang How is this done with oss-fuzz?

OSS-Fuzz works by running at HEAD, so that wasn't a consideration there.

`last-stable` as a place that doesn't necessariy follow `main` but as somewhere that we can cherry-pick to SGTM. 

One thing I see other projects do is also provide an option to view the "main" documentation as well through a dropdown. I think this will be useful to both users and ourselves (to validate the documentation we're releasing looks good). 

> 
> > 2). I am assuming the github page should be https://google.github.io/osv-scanner please let me know if a different url is preferred.
> 
> That url looks good to me.

+1.

**Comment by hayleycd** [comment]
> One thing I see other projects do is also provide an option to view the "main" documentation as well through a dropdown. I think this will be useful to both users and ourselves (to validate the documentation we're releasing looks good).

I'll look into that.

**Comment by hayleycd** [comment]
We can potentially organize our docs carefully and use [labels](https://just-the-docs.github.io/just-the-docs/docs/ui-components/labels/) and/or [callouts](https://just-the-docs.github.io/just-the-docs/docs/ui-components/callouts/) to indicate documentation for things that are upcoming, while building from a single branch. 

Thoughts?

**Comment by hayleycd** [comment]
So I have been thinking about this for a bit and experimenting with options. Will you take a look at [this](https://hayleycd.github.io/doc_poc/) and let me know what you think?

TL;DR of the link above is that there is a "current" version of the docs and a "working" version. They are both always deployed in the docs website, but the working docs are excluded from the navigation and you could only get there by knowing the url.  Callouts are at the top of the page to tell users they are in the wrong place. There is then a link back to the current docs.

**Comment by another-rex** [comment]
Just had a look, I think that's pretty good system, though I am a bit confused on what the difference is between the upcoming doc and the working doc?

Looking at this issue https://github.com/just-the-docs/just-the-docs/issues/728 it seems like they haven't implemented the versioning that Oliver describes yet.

**Comment by hayleycd** [comment]
> Just had a look, I think that's pretty good system, though I am a bit confused on what the difference is between the upcoming doc and the working doc?

@another-rex I had used the word "upcoming" before deciding I liked "working" better.

**Comment by hayleycd** [comment]
We could set up something like what is mentioned [here](https://github.com/just-the-docs/just-the-docs/issues/728#issuecomment-1176560963) but I'm not sure if that is necessary given the size of the project. Happy to do something like that if the team thinks it is worth it in the long term.

**Comment by another-rex** [comment]
I personally don't think it is worth it to have versioning for the docs, osv-scanner isn't maintaining a lot of different versions/branches at the same time, just the latest release version, reducing the usefulness of having docs for past versions.

If someone really needs to look at the docs for a specific version and can't upgrade their osv-scanner install, they can always jump to the release in git history and read it there.

**Comment by oliverchang** [comment]
+1 not worth having something complicated to get the version dropdowns. What you suggested @hayleycd sounds good to me!

**Comment by hayleycd** [comment]
#203 merged. Closing issue.
